driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java

```diff
@@ -46,7 +48,7 @@ public void compress(final List<ByteBuf> source, final BsonOutput target) {
 
         try {
             byte[] out = new byte[(int) Zstd.compressBound(uncompressedSize)];
-            int compressedSize = (int) Zstd.compress(out, singleByteArraySource, Zstd.maxCompressionLevel());
+            int compressedSize = (int) Zstd.compress(out, singleByteArraySource, DEFAULT_COMPRESSION_LEVEL);
```

Comment:
```suggestion
            int compressedSize = (int) Zstd.compress(out, singleByteArraySource, Zstd.defaultCompressionLevel());
```

Reply:
The `defaultCompressionLevel` was not added until a later version of zstd-jni. I considered upgrading to the latest zstd-jni but thought you might want to minimize additional changes. Let me know if you're ok with upgrading. You could consider it a backport of https://jira.mongodb.org/browse/JAVA-4965  :-}

Reply:
Let's not do that.  It could easily break people's applications